include/range/v3/utility/swap.hpp

```diff
@@ -42,10 +53,26 @@ namespace ranges
         /// \cond
         namespace adl_swap_detail
         {
-            using std::swap;
+
+            // The following implementation of swap is adapted from libc++:
+            template <class T>
+            RANGES_CXX14_CONSTEXPR
+            typename std::enable_if<
+                std::is_move_constructible<T>::value &&
+                std::is_move_assignable<T>::value
+             >::type
+            swap(T& x, T& y)
+            noexcept(std::is_nothrow_move_constructible<T>::value &&
+                     std::is_nothrow_move_assignable<T>::value)
+            {
+                T t(std::move(x));
+                x = std::move(y);
+                y = std::move(t);
+            }
```

Comment:
This just doesn't work. It's perfectly acceptable for users to explicitly specialize `swap` on user-defined types in namespace `std`. Routing around `std::swap` just because it happens not to be `constexpr` is fundamentally broken.